pandas/tests/dtypes/test_cast.py

```diff
@@ -407,3 +408,12 @@ def test_period_dtype(self):
                        np.dtype('datetime64[ns]'), np.object, np.int64]:
             assert find_common_type([dtype, dtype2]) == np.object
             assert find_common_type([dtype2, dtype]) == np.object
+
+    @pytest.mark.parametrize('datum1', [1, 2., "3", (4, 5), [6, 7], None])
+    @pytest.mark.parametrize('datum2', [8, 9., "10", (11, 12), [13, 14], None])
+    def test_cast_1d_array(self, datum1, datum2):
```

Comment:
are there fail cases? iow where this routine raises?

Reply:
Not once you remove support for non-``object`` dtypes (which is what you just asked me to do).

(well, as long as ``data`` has a ``len()`` and is iterable...)

Reply:
right so can you add a test that fails for scalars then

Reply:
(done)